paimon-common/src/main/java/org/apache/paimon/fileindex/bitmap/BitmapIndexResultLazy.java

```diff
@@ -38,9 +38,13 @@ public boolean remain() {
     public FileIndexResult and(FileIndexResult fileIndexResult) {
         if (fileIndexResult instanceof BitmapIndexResultLazy) {
             return new BitmapIndexResultLazy(
-                    () ->
-                            RoaringBitmap32.and(
-                                    get(), ((BitmapIndexResultLazy) fileIndexResult).get()));
+                    () -> {
+                        if (!remain()) {
```

Comment:
thanks, rollbacked.

Reply:
This can indeed be optimized, just return SKIP instead of get();

Reply:
I take a look the file index framework, it will fast skip when current predicate is fasle. See https://github.com/apache/paimon/blob/32d05f31e1aae54e7d7504fc5f66cbde4a111a91/paimon-common/src/main/java/org/apache/paimon/fileindex/FileIndexPredicate.java#L164